near-sdk-macros/src/core_impl/event/mod.rs

```diff
@@ -0,0 +1,75 @@
+use proc_macro::TokenStream;
+
+use proc_macro2::Span;
+use quote::quote;
+use syn::{parse_quote, ItemEnum};
+
+/// This attribute macro is used on a enum and its implementations
+/// to generate the necessary code to format standard event logs.
+///
+/// This macro will generate code to load and deserialize state if the `self` parameter is included
+/// as well as saving it back to state if `&mut self` is used.
+///
+/// For parameter serialization, this macro will generate a struct with all of the parameters as
+/// fields and derive deserialization for it. By default this will be JSON deserialized with `serde`
+/// but can be overwritten by using `#[serializer(borsh)]`.
+///
+/// `#[near_bindgen]` will also handle serializing and setting the return value of the
+/// function execution based on what type is returned by the function. By default, this will be
+/// done through `serde` serialized as JSON, but this can be overwritten using
+/// `#[result_serializer(borsh)]`.
+///
+/// # Examples
+///
+/// ```ignore
+/// use near_sdk::{near_events, Event};
+///
+/// #[near_events]
+/// pub enum MyEvents {
+///    #[event_standard("swap_standard")]
+///    #[event_version("1.0.0")]
+///    Swap { token_in: AccountId, token_out: AccountId, amount_in: u128, amount_out: u128 },
+///
+///    #[event_standard("string_standard")]
+///    #[event_version("2.0.0")]
+///    StringEvent(String),
+///
+///    #[event_standard("empty_standard")]
+///    #[event_version("3.0.0")]
+///    EmptyEvent
+/// }
+///
+/// #[near_bindgen]
+/// impl Contract {
+///     pub fn some_function(&self) {
+///         Event::emit {
```

Comment:
How does this `Event::emit {` work? Shouldn't it be a function call with `()` instead of `{}`?

Reply:
yep, I will change this